core/sparqlbuilder/src/main/java/org/eclipse/rdf4j/sparqlbuilder/constraint/Operation.java

```diff
@@ -23,7 +23,9 @@
 	Operation(SparqlOperator operator, int operandLimit) {
 		super(operator);
 		this.operandLimit = operandLimit;
-//		parenthesize();
+		if (operator.toString().equals("AND")) {
```

Comment:
Using toString() and then .equals() is typically very slow. Try to use the enums instead: ConnectiveOperator.AND

Reply:
Thanks for the suggestion! Done the necessary changes.